File: Ora.GameManaging.Server/Models/RoleStatusModel.cs
namespace Ora.GameManaging.Server.Models
{
    public class RoleStatusModel
    {
        public required string RoleName { get; set; }
        public int Health { get; set; }
        public int AbilityCount { get; set; }
        public int SelfAbilityCount { get; set; }
        public bool HasNightAbility { get; set; }
        public bool HasDayAbility { get; set; }
        public bool CanSpeak { get; set; }
        public bool DarkSide { get; set; }
        public required string Abilities { get; set; }
    }
}

File: client-app/src/app/models/RoleStatusModel.ts
import { AbilityModel } from "./AbilityModel";

export interface RoleStatusModel {
    roleName: string;
    health: number;
    abilityCount: number;
    selfAbilityCount: number;
    hasNightAbility: boolean;
    hasDayAbility: boolean;
    canSpeak: boolean;
    darkSide: boolean;
    abilities?: AbilityModel[];
}